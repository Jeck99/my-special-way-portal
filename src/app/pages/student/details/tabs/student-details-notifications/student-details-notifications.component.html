<ng-template #rehabEnable>
  <div class="table">
    <div class="row">
      <div class="cell">
        <p>גמילה</p>
      </div>
      <div class="cell">
        <mat-slide-toggle
          [checked]="rehabToggleMode"
          (change)="onToggleChanged('rehab', $event.checked)"></mat-slide-toggle>
      </div>
    </div>
    <div class="row">
      <div class="cell">
        ימים א-ה
      </div>
      <div class="cell">
        <p class="value">08:30, 09:30, 10:30, 11:30, 12:30, 13:30, 14:30, 15:30, 16:30</p>
      </div>
    </div>
    <div class="row">
      <div class="cell">
        יום ו
      </div>
      <div class="cell">
        <p class="value">08:30, 09:30, 10:30, 11:30, 12:30, 13:30</p>
      </div>
    </div>
  </div>
</ng-template>
<ng-template #rehabDisabled>
  <div class="table">
    <div class="row">
      <div class="cell">
        <p>תזכורת גמילה</p>
      </div>
      <div class="cell">
        <mat-slide-toggle
          [checked]="rehabToggleMode"
          (change)="onToggleChanged('rehab', $event.checked)"></mat-slide-toggle>
      </div>
    </div>
    <div class="row">
      <div class="cell">
        בחר יום ושעה
      </div>
    </div>
  </div>
</ng-template>


<ng-template #medicineEnable>
  <div class="table">
    <div class="row">
      <div class="cell">
        <p>תרופה</p>
      </div>
      <div class="cell">
        <mat-slide-toggle
          [checked]="medicineToggleMode"
          (change)="onToggleChanged('medicine', $event.checked)"></mat-slide-toggle>
      </div>
    </div>
    <div class="row">
      <div class="cell">
        ימים א-ה
      </div>
      <div class="cell">
        <p class="value">08:30, 09:30</p>
      </div>
    </div>
    <div class="row">
      <div class="cell">
        יום ו
      </div>
      <div class="cell">
        <p class="value">10:30</p>
      </div>
    </div>
  </div>
</ng-template>
<ng-template #medicineDisabled>
  <div class="table">
    <div class="row">
      <div class="cell">
        <p>תזכורת תרופה</p>
      </div>
      <div class="cell">
        <mat-slide-toggle
          [checked]="medicineToggleMode"
          (change)="onToggleChanged('medicine', $event.checked)"></mat-slide-toggle>
      </div>
    </div>
    <div class="row">
      <div class="cell">
        בחר יום ושעה
      </div>
    </div>
  </div>
</ng-template>


<div class="msw-student-details-notifications">
  <mat-card class="student-notifications rehab selected-radius" (click)="updateStudentReminder('rehab')">
    <ng-container
      *ngIf="rehabToggleMode; then rehabEnable; else rehabDisabled">
    </ng-container>
  </mat-card>
  <br>
  <mat-card class="student-notifications medicine selected-radius" (click)="updateStudentReminder('medicine')">
    <ng-container
      *ngIf="medicineToggleMode; then medicineEnable; else medicineDisabled">
    </ng-container>
  </mat-card>
</div>

