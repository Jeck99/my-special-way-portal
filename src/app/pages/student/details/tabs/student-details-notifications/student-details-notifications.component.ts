import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

@Component({
  selector: 'app-student-details-notifications',
  templateUrl: './student-details-notifications.component.html',
  styleUrls: ['./student-details-notifications.component.scss'],
})
export class StudentDetailsNotificationsComponent implements OnInit {

  sub: any;
  idOrNew: string;

  constructor(
    private route: ActivatedRoute,
  ) { }

  ngOnInit(): void {
    if (this.route && this.route.parent) {
      this.sub = this.route.parent.params.subscribe((params) => {
        this.idOrNew = params.idOrNew;
      });
    }
  }
}
