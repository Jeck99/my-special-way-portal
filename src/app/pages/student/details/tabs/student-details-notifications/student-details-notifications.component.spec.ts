import { TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { Platform } from '@angular/cdk/platform';
import { Observable } from 'rxjs-compat';
import { ActivatedRoute, Router, RouterModule } from '@angular/router';
import { StudentDetailsNotificationsComponent } from './student-details-notifications.component';
import { StudentDetailsComponent } from '../../student-details.component';
import { MatDialog } from '@angular/material';
let studentReminderDialogMock: Partial<MatDialog>;

describe('Student Details Notifications Component', () => {
  beforeEach(async () => {
    studentReminderDialogMock = {
      open: jest.fn().mockReturnValue({
        afterClosed: jest.fn().mockReturnValue(Observable.of(true)),
      }),
    };

    TestBed.configureTestingModule({
      imports: [RouterModule.forRoot([])],
      declarations: [StudentDetailsComponent, StudentDetailsNotificationsComponent],
      providers: [
        {
          provide: Router,
          useValue: {
            forRoot: jest.fn(),
          },
        },
        Platform,
        {
          provide: ActivatedRoute,
          useValue: {
            parent: {
              params: Observable.of({ idOrNew: '_new_' }),
            },
          },
        },
        { provide: MatDialog, useValue: studentDialogMock },
      ],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  describe('with _new_ student path', () => {
    beforeEach(async () => {});

    it('should render the component as described in snapshot', () => {
      const fixture = TestBed.createComponent(StudentDetailsNotificationsComponent);
      expect(fixture).toMatchSnapshot();
    });

    it('should subscribe to params onInit', () => {
      const activatedRouteMock = {
        parent: {
          params: {
            subscribe: jest.fn().mockImplementationOnce((callback) => callback({})),
          },
        },
      } as never;
      // given
      const studentDetailsNotification = new StudentDetailsNotificationsComponent(activatedRouteMock);

      // when
      studentDetailsNotification.ngOnInit();

      // then
      // tslint:disable-next-line:no-non-null-assertion test case only
      const subscribeMock = (activatedRouteMock as ActivatedRoute).parent!.params.subscribe as jest.Mock;
      expect(subscribeMock).toHaveBeenCalled();
    });
  });
});
