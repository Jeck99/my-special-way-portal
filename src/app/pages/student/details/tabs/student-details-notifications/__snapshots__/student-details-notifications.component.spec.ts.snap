// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Details Notifications Component with _new_ student path should render the component as described in snapshot 1`] = `
<app-student-details-notifications
  dialog="undefined"
  medicineToggleMode="false"
  rehabToggleMode="false"
  route={[Function Object]}
>
  <div
    class="msw-student-details-notifications"
  >
    <mat-card
      class="student-notifications rehab selected-radius"
    >
      
    </mat-card>
    <br />
    <mat-card
      class="student-notifications medicine selected-radius"
    >
      
    </mat-card>
  </div>
</app-student-details-notifications>
`;
